### PR TITLE
[Merged by Bors] - feat: forward-port leanprover-community/mathlib#18359

### PR DESCRIPTION
**Pull request by eric-wieser**
Also fixes a lemma statement to use `%` notation like it did in Lean3

---

- [x] depends on: #2178
- [x] depends on: #2542

[Open in Gitpod](https://gitpod.io/from-referrer/)

* leanprover-community/mathlib#18359

* [`logic.equiv.fin`@`2445c98ae4b87eabebdde552593519b9b6dc350c`..`bd835ef554f37ef9b804f0903089211f89cb370b`](https://leanprover-community.github.io/mathlib-port-status/file/logic/equiv/fin?range=2445c98ae4b87eabebdde552593519b9b6dc350c..bd835ef554f37ef9b804f0903089211f89cb370b)
* [`data.int.order.basic`@`b86832321b586c6ac23ef8cdef6a7a27e42b13bd`..`bd835ef554f37ef9b804f0903089211f89cb370b`](https://leanprover-community.github.io/mathlib-port-status/file/data/int/order/basic?range=b86832321b586c6ac23ef8cdef6a7a27e42b13bd..bd835ef554f37ef9b804f0903089211f89cb370b)
* [`data.nat.basic`@`2609ad04e2b77aceb2200e273daf11cc65856fda`..`bd835ef554f37ef9b804f0903089211f89cb370b`](https://leanprover-community.github.io/mathlib-port-status/file/data/nat/basic?range=2609ad04e2b77aceb2200e273daf11cc65856fda..bd835ef554f37ef9b804f0903089211f89cb370b)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#2178
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#2178~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/Logic/Equiv/Fin.lean [hunk 1]
```diff
@@ -491,6 +492,43 @@ def finProdFinEquiv : Fin m × Fin n ≃ Fin (m * n)
...
+#align nat.div_mod_equiv Nat.divModEquiv
+
+/-- The equivalence induced by `a ↦ (a / n, a % n)` for nonzero `n`.
+See `Int.ediv_mod_unique` for a similar propositional statement. -/
+@[simps]
+def Int.divModEquiv (n : ℕ) [NeZero n] : ℤ ≃ ℤ × Fin n where
+  -- TODO: could cast from int directly if we import `data.zmod.defs`, though there are few lemmas
+  -- about that coercion.
+  toFun a := (a / n, ↑(a.natMod n))
+  -- porting note: need explicit `Nat.cast` else we get `Int.ofNat` as the coercion instead.
```
Comment (eric-wieser) [inline Mathlib/Logic/Equiv/Fin.lean:517]: [Zulip thread](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/simp.20normal.20form.3A.20ofNat.20or.20Nat.2Ecast.3F/near/338423289)

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#2178~~
* ~~leanprover-community/mathlib4#2542~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by joneugster** [comment]
This looks good to me, just fixed a few typos.

File: Mathlib/Logic/Equiv/Fin.lean (hunk 1 above)
Comment (eric-wieser) [reply Mathlib/Logic/Equiv/Fin.lean:517]: #2878 will fix this.

**Review by Parcly-Taxel — APPROVED** [review]

**Comment by RemyDegenne** [comment]
Thanks
bors r+

**Comment by eric-wieser** [comment]
bors merge

**Pull request closed without merging**